sound/soc/sof/intel/hda-dai.c

```diff
@@ -585,7 +481,7 @@ void hda_set_dai_drv_ops(struct snd_sof_dev *sdev, struct snd_sof_dsp_ops *ops)
 			if (strstr(ops->drv[i].name, "iDisp") ||
 			    strstr(ops->drv[i].name, "Analog") ||
 			    strstr(ops->drv[i].name, "Digital"))
-				ops->drv[i].ops = &ipc4_hda_dai_ops;
+				ops->drv[i].ops = &hda_dai_ops;
```

Comment:
this is a bit confusion, if the ops are the same between IPC and IPC4 why do wee need the switch()?